lightning/src/routing/network_graph.rs

```diff
@@ -428,6 +428,70 @@ impl Readable for NodeInfo {
 	}
 }
 
+/// Allows for updating user's network metadata.
+pub trait UpdateMetadata {
```

Comment:
Seems to me this trait is trying to encompass both inputs consuming to compute score (`add_failed_route`) and scores consumption (`get_channel_fee_penalty`). Second is likely to be consumed by Router and any-kind of peer selection, but first-one is really tied about the origin of data consumed. 

If it's event-based, like receiving a PaymentSuccess or PaymentFailed, client is going to call a `add_failed_route` method-like. If it's self-running, like a thread probing/tracking peer liveness,, you just call something like `spawn_liveness_thread`. So I would favor dissociate those traits even if their are implemented on the same default struct.

Reply:
I can disassociate these traits, I think that makes sense. Maybe the NetworkGraph <---> User Metadata connection should be called UpdatesMetadata, and the User metadata <---> router should be called ScoresChannels. The user would implement both traits, and the network would, by default, send certain data to the traits, and the router would by, default, ask for channel scores when conducting its A* search.